support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/ticket/accesstoken/OAuthAccessTokenIdExtractor.java

```diff
@@ -0,0 +1,88 @@
+package org.apereo.cas.ticket.accesstoken;
+
+import org.apereo.cas.services.RegisteredService;
+import org.apereo.cas.services.RegisteredServiceAccessStrategyUtils;
+import org.apereo.cas.services.RegisteredServiceCipherExecutor;
+import org.apereo.cas.services.ServicesManager;
+import org.apereo.cas.support.oauth.services.OAuthRegisteredService;
+import org.apereo.cas.support.oauth.util.OAuth20Utils;
+import org.apereo.cas.util.crypto.CipherExecutor;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+import org.jose4j.jwt.JwtClaims;
+
+import java.io.Serializable;
+import java.util.Optional;
+
+/**
+ * This is {@link OAuthAccessTokenIdExtractor}.
+ *
+ * @author charlibot
+ * @since 6.1.0
+ */
+@Slf4j
+@RequiredArgsConstructor
+@Getter
+public class OAuthAccessTokenIdExtractor {
```

Comment:
Would be preferable if this could implement an interface that defines the extraction contract.